libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/LargeCausalIndividualView/LargeCausalIndividualChart.tsx

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IComboBoxOption, IComboBox, Stack } from "@fluentui/react";
+import {
+  IGenericChartProps,
+  ISelectorConfig,
+  MissingParametersPlaceholder,
+  defaultModelAssessmentContext,
+  ModelAssessmentContext,
+  rowErrorSize,
+  OtherChartTypes,
+  IScatterPoint,
+  IHighchartsConfig,
+  ChartTypes,
+  instanceOfHighChart,
+  getScatterOption,
+  ICausalAnalysisData
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import React from "react";
+
+import { causalIndividualChartStyles } from "../CausalIndividualChart.styles";
+import { generateDefaultChartAxes } from "../generateChartProps";
+
+import { getLocalCausalFromSDK } from "./getOnScatterPlotPointClick";
+import {
+  getInitialSpec,
+  ILargeCausalIndividualChartProps,
+  ILargeCausalIndividualChartState
+} from "./ILargeCausalIndividualChartSpec";
+import { LargeCausalIndividualChartArea } from "./LargeCausalIndividualChartArea";
+import { LargeCausalIndividualChartLegend } from "./LargeCausalIndividualChartLegend";
+import {
+  generateHighChartConfigOverride,
+  getBubblePlotData,
+  getErrorMessage,
+  getNewChartProps,
+  getNewSelections,
+  getTemporaryPoint,
+  instanceOfLocalCausalData,
+  selectPointFromChartLargeData,
+  selectPointFromDropdownIntl,
+  shouldUpdateHighchart
+} from "./largeCausalIndividualChartUtils";
+
+export class LargeCausalIndividualChart extends React.PureComponent<
+  ILargeCausalIndividualChartProps,
+  ILargeCausalIndividualChartState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+  private changedKeys: string[] = [];
```

Comment:
I might have missed something here. Is this `changedKeys` been updated somewhere in this component other than passing it to `shouldUpdateHighchart`?

Reply:
Thanks Tong, since we use this in recursive function I was using this as a class variable. I moved this code  to core package in this PR. I will check if I can use this as a const in my coming PR.